libs/libarchfpga/src/physical_types_util.h

```diff
@@ -0,0 +1,132 @@
+#ifndef PHYSICAL_TYPES_UTIL_H
+#define PHYSICAL_TYPES_UTIL_H
+
+#include "physical_types.h"
+
+/********************************************************************
+ *                                                                  * 
+ *  Physical types utility functions                                *
+ *                                                                  *
+ *  This source file contains several utilities that enable the     *
+ *  interaction with the architecture's physical types.             *
+ *  Mainly, the two classes of objects accessed by the utility      *
+ *  functions in this file are the following:                       *
+ *    - physical_tile_type                                          *
+ *    - logical_block_tpye                                          *
+ *                                                                  *
+ *  All the following utilities are intended to ease the            *
+ *  developement to access the above mentioned classes and perform  *
+ *  some required operations with their data.                       *
+ *                                                                  *
+ *  Please classify such functions in this file                     *
+ *                                                                  *
+ * ******************************************************************/
+
+bool is_opin(int ipin, t_physical_tile_type_ptr type);
+bool is_input_type(t_physical_tile_type_ptr type);
+bool is_output_type(t_physical_tile_type_ptr type);
+bool is_io_type(t_physical_tile_type_ptr type);
+
+//Returns the corresponding physical pin based on the input parameters:
+// - physical_tile
+// - relative_pin: this is the pin relative to a specific sub tile
+// - capacity location: absolute sub tile location
+int get_physical_pin_from_capacity_location(t_physical_tile_type_ptr physical_tile, int relative_pin, int capacity_location);
+
+//Returns a pair consisting of the absolute capacity location relative to the pin parameter
+//and the relative pin within the sub tile
+std::pair<int, int> get_capacity_location_from_physical_pin(t_physical_tile_type_ptr physical_tile, int pin);
```

Comment:
Again an example here would help I think (can define an example and use it for a few of these routines.  I'm not quite sure what "capacity location" means so precisely defining that (or giving an example) would help.

Reply:
Done